cmake/pcl_find_vtk.cmake

```diff
@@ -102,28 +102,20 @@ if (vtkMissingComponents)
   message(WARNING "Missing vtk modules: ${vtkMissingComponents}")
 endif()
 
-if(WITH_QT)
-  if(VTK_VERSION VERSION_LESS 9.0)
-    if(";${VTK_MODULES_ENABLED};" MATCHES ";vtkGUISupportQt;" AND ";${VTK_MODULES_ENABLED};" MATCHES ";vtkRenderingQt;")
-      set(HAVE_QVTK TRUE)
-      #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
-      #Pre vtk 9.0, all vtk libraries are linked into pcl_visualizer.
-      #Subprojects can link against pcl_visualizer and directly use VTK-QT libraries.
-      list(APPEND PCL_VTK_COMPONENTS vtkRenderingQt vtkGUISupportQt)
-    else()
-      unset(HAVE_QVTK)
-    endif()
-  else()
-	if(";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";GUISupportQt;" AND ";${VTK_AVAILABLE_COMPONENTS};" MATCHES ";RenderingQt;")
-      set(HAVE_QVTK TRUE)
-      #PCL_VTK_COMPONENTS is used in the PCLConfig.cmake to refind the required modules.
-      #Post vtk 9.0, only required libraries are linked against pcl_visualizer.
-      #Subprojects need to manually link to VTK-QT libraries.
-      list(APPEND PCL_VTK_COMPONENTS RenderingQt GUISupportQt)
-    else()
-      unset(HAVE_QVTK)
-    endif()
-  endif()
+if(";${VTK_MODULES_ENABLED};" MATCHES ";vtkGUISupportQt;" AND ";${VTK_MODULES_ENABLED};" MATCHES ";vtkRenderingQt;")
```

Comment:
This is for https://cmake.org/cmake/help/latest/policy/CMP0054.html I think. Do we support CMake pre 3.1 ? :)

Reply:
Nop, currently 3.5 is the minimum version (and 3.10 upcoming):

https://github.com/PointCloudLibrary/pcl/blob/f8a48ddd61141601c6b718a4d15d9040ca0e1662/CMakeLists.txt#L2

As these are always string comparism (because of the `;`) CMP0054 is not relevant on this check here.

Reply:
Isn't the addition of `;` to prevent double expansion of VTK_MODULES_ENABLED

Would a simple `if(VTK_MODULES_ENABLED MATCHES "vtkGUISupportQt" AND ...)` suffice?

Reply:
Seems so. Just pushed changes and lets see how CI's goes.

Reply:
Not really, the `;` are there because it is a list and CMake doesn't have a `variable CONTAINS xyz` command.

Therefore changing:
```CMake
if(";${VTK_MODULES_ENABLED};" MATCHES ";vtkGUISupportQt;")
```

to

```CMake
if(VTK_MODULES_ENABLED MATCHES "vtkGUISupportQt")
```

would lead to a match of e.g. a module named `vtkGUISupportQtAdvanced`

Simplifing it to
```CMake
if(VTK_MODULES_ENABLED MATCHES ";vtkGUISupportQt;")
```

could lead to a missing detection of a module in case `vtkGUISupportQt` is the first item within `VTK_MODULES_ENABLED`

You can try it by adding this line before the checks:
```CMake
unset(VTK_MODULES_ENABLED)
list(APPEND VTK_MODULES_ENABLED "GUISupportQt" "vtkGUISupportQtAdvanced" "vtkRenderingQtAdvanced" "RenderingQt")
```

On current master this should go into the legacy VTK code. After current changes here it goes into new VTK code. I know the case is a bit constructed, but just want to tell you, that this changes the behavior.

Reply:
I searched for this again and we can use IN_LIST https://cmake.org/cmake/help/v3.3/command/if.html.

Hopefully it works with your suggested cases as well @SunBlack.

Reply:
Cool, didn't knew that this command exists :-)

Reply:
CMake is a leviathan. The more you know, the more sensible madness seems